lib/ex_crypto.ex

```diff
@@ -36,6 +36,33 @@ defmodule ExCrypto do
     {kind, Exception.normalize(kind, error), System.stacktrace}
   end
 
+  @aes_block_size 16
+  @iv_bit_length 128
+
+  defp key_and_iv_for_algorithm(raw_key, raw_iv, algorithm) do
+    bits = algorithm_bits(algorithm)
+    {iv, _} = bitstring_size(raw_iv, @iv_bit_length)
+    {key, _} = bitstring_size(raw_key, bits)
+    {key, iv}
+  end
+
+  defp test_key_and_iv_against_algorithm(key, iv, algorithm) do
+    iv_size = bit_size(iv)
+    key_size = bit_size(key)
+    if iv_size != @iv_bit_length || key_size != algorithm_bits(algorithm) do
+      raise "IV must be exactly 128 bits and key must be exactly #{algorithm_bits(algorithm)} bits, or use encrypt!/decrypt! method"
+    end
+  end
+
+  defp algorithm_bits(algorithm) do
```

Comment:
Hah, yeah, maybe got ahead of myself since we're technically only supporting 256 bit now, but you support generating keys for other algo-bit-lengths, and I thought shouldn't hardcode the detection to 256, but just trust that all the algorithms have their bitlength in the algo, so we can extract an integer from the atom using this and then use that to check that their key is exactly the right length (I got bitten multiple times by having wrong key length and IV and erlangs errors don't give you any indication that's the problem, which is frustrating).